### PR TITLE
Fix deflater stream in upload data

### PR DESCRIPTION
**Pull request by grooze**

### CODE REVIEW

File: appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java [hunk 1]
```diff
@@ -581,6 +590,8 @@ protected void uploadData() {
                         return;
                     }
                     uploadDataStream = null;
+                    outputStream = null;
+                    writeCount = 0;
                 }
             }
```
Comment (akondratyev) [inline appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java:596]: may be there's sense to do 2 things: split logic of chunking and uploading data 
and write a couple of tests for chunk?

Comment (grooze) [reply appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java:596]: There is only chunking and compressing data, uploading in UploadCacheTask, I don't understand - what's problem?

Comment (akondratyev) [reply appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java:596]: I don't see any problem now
But if we split code in more methods and **test** chunking - we will be more confident in implementation

Comment (grooze) [reply appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java:596]: Unit tests not worked on Android at all due to very old code (Espresso not worked with last build tools). We need to refactor core test classes

Comment (AntonGabov) [reply appmetr/src/main/java/com/appmetr/android/internal/AppMetrTrackingManager.java:596]: I hope it will work as we couldn't test this code :)

**Review by AntonGabov — APPROVED** [review]

**Review by akondratyev — APPROVED** [review]

**Pull request merged**